### DISCUSSION TITLE
Avoid downstream filtering of compound hets for the same family

### DISCUSSION THREAD

**Discussion by NLSVTN** [broadinstitute/seqr#2042] [Q&A] [answered]
The results returned by compound het query can return variants from different families:

https://github.com/broadinstitute/seqr/blob/master/seqr/utils/elasticsearch/es_search.py#L751

and if the grouping contains only one variant they will be filtered out:

https://github.com/broadinstitute/seqr/blob/master/seqr/utils/elasticsearch/es_search.py#L809

Do you know if it could be possible to somehow eliminate this particular step and instead making Elasticsearch to do that? We really need it because we split compound het query into 2: genes bucketing and then retrieving variants by the user selecting individual genes; this improves speed and prevents server crashes. In case when we select 'all families' there are many genes where individual gene query returns nothing due to this grouping and filtering by single family.

**Comment by hanars** [comment, marked as answer]
We do not support recessive/ compound het search across more than a couple families in seqr, so this is not a use case that we will be fixing. 

This is not possible to do in elasticsearch as checking phasing within families is not something its filtration can handle
